Hublsoft.Net.LoginLogout.DataAccess/RegisteredUsersRepository.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using MySql;
+using MySql.Data;
+using MySql.Data.MySqlClient;
+
+namespace Hublsoft.Net.LoginLogout.DataAccess
+{
+    public class RegisteredUsersRepository : BaseRepository, IRegisteredUsersRepository
+    {
+        public RegisteredUsersRepository(IOptions<DatabaseOptions> databaseOptions) : base(databaseOptions) { }
+
+        public async Task<int> GetIdByEmailAddressAsync(string emailAddress)
+        {
+            var id = 0;
+
+            using (var con = new MySqlConnection(base.ConnectionString))
+            {
+                await con.OpenAsync();
+
+                string stm = $"SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED ; SELECT Id FROM RegisteredUsers WHERE EmailAddress = '{emailAddress}' LIMIT 1 ; SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ ;";
```

Comment:
Are injection attacks a concern here?

Reply:
Agreed, I refrained from doing that at this stage so that I could quickly change the code whilst things weren't working!

Not for EmailAddress because I added the `[DataType(DataType.EmailAddress)]` annotation which prohibits anything that isn't an email address during data binding. However, there is no validation on Password. Based on the tech spec I wrote, this *assumes* that the consuming application will hash the password. That said, I should probably add a regular expression annotation

Reply:
Yeah, I'm always more defensive with these things. As much as possible allow the code to run, but at least to checks and notify the application (warnings vs errors etc)